Report all probes

speshak opened the pull request:
Fixes #45

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/thorrak/brewpi-esp8266?pullRequest=46) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.io/thorrak/brewpi-esp8266?pullRequest=46) before we can accept your contribution.<br/>**0** out of **2** committers have signed the CLA.<br/><br/>:x: speshak<br/>:x: thorrak<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/thorrak/brewpi-esp8266?pullRequest=46) it.</sub>

speshak commented:
I'm opening this now as more of an RFC before I go further.  There are a couple outstanding issues that at least need discussed:

- The "keywords" I chose for extending the PiLink protocol.  I used `T` and `o`, but there might be better options.
- Being more defensive in how the custom names are stored.  If you were being malicious, you could overwrite some of the other system data by claiming a device address of `/mdns.txt`
- Is there a reason why `parseJsonToken` [eats spaces](https://github.com/thorrak/brewpi-esp8266/blob/esp32/src/PiLink.cpp#L763)?  This means that no probe name can have a space.  That could be a reasonable trade off, especially if other code paths depend on that behavior, but it was a bit surprising when I was testing.

speshak commented:
I've answered my own question about `parseJsonToken`.  That function is used for both JSON keys and values, and spaces (and the other skipped chars) aren't valid in a key.

lbussy commented:
![](https://imgflip.com/s/meme/Leonardo-Dicaprio-Cheers.jpg)

speshak commented:
I fixed the issue with possibly squashing other SPIFFS content, and added another PiLink command (`p`) for dumping the list of configured device names.  I think this is ready for review.

thorrak commented on src/DeviceNameManager.cpp:
```diff
@@ -0,0 +1,149 @@
+  char filename[32];
+  DeviceNameManager::deviceNameFilename(filename, device);
+
+  File f = SPIFFS.open(filename, "w");
```
At some point (later!) we should probably revisit how the device definitions are written out and see if they can also be refactored into JSON writes. Project for later!
